tests/unit/helpers/questionnaires.test.js

```diff
@@ -256,42 +266,84 @@ describe('getUserQuestionnaires', () => {
         { query: 'findOne', args: [{ _id: courseId }] },
         { query: 'populate', args: [{ path: 'slots', select: '-__v -createdAt -updatedAt' }] },
         { query: 'populate', args: [{ path: 'slotsToPlan', select: '_id' }] },
+        {
+          query: 'populate',
+          args: [{ path: 'subProgram', select: 'program', populate: { path: 'program', select: '_id' } }],
+        },
         { query: 'lean', args: [{ virtuals: true }] },
       ]
     );
     SinonMongoose.calledOnceWithExactly(
-      findOneQuestionnaire,
+      findQuestionnaires,
       [
-        { query: 'findOne', args: [{ type: EXPECTATIONS, status: PUBLISHED }, { type: 1, name: 1 }] },
+        {
+          query: 'find',
+          args: [
+            {
+              type: { $in: [EXPECTATIONS, SELF_POSITIONNING] },
+              $or: [{ program: { $exists: false } }, { program: course.subProgram.program._id }],
+              status: PUBLISHED,
+            },
+            { type: 1, name: 1 },
+          ],
+        },
         {
           query: 'populate',
           args: [{
             path: 'histories',
-            match: { course: course._id, user: credentials._id },
+            match: {
+              course: course._id,
+              user: credentials._id,
+              $or: [{ timeline: { $exists: false } }, { timeline: START_COURSE }],
+            },
             options: { requestingOwnInfos: true },
-            select: { _id: 1 },
+            select: { _id: 1, timeline: 1 },
           }],
         },
         { query: 'lean', args: [{ virtuals: true }] },
       ]
     );
   });
 
-  it('should return an empty array if expectations questionnaire is already answered', async () => {
+  it('should return an empty array if not started and published questionnaires are already answered', async () => {
```

Comment:
je trouve ce titre bizarre on comprend pas trop le not started, il manque "course", non ?

Reply:
Oui j'ai clarifié tous les intitulés